runelite-client/src/main/java/net/runelite/client/plugins/opponentinfo/OpponentInfoPlugin.java

```diff
@@ -134,33 +138,48 @@ else if (worldType.contains(WorldType.LEAGUE))
 	@Subscribe
 	public void onInteractingChanged(InteractingChanged event)
 	{
-		if (event.getSource() != client.getLocalPlayer())
-		{
-			return;
-		}
-
-		Actor opponent = event.getTarget();
+		final Actor player = client.getLocalPlayer();
+		final Actor source = event.getSource();
+		final Actor target = event.getTarget();
 
-		if (opponent == null)
+		if (source == player)
 		{
-			lastTime = Instant.now();
-			return;
+			if (target != null)
+			{
+				lastOpponent = target;
+				lastTime = null;
+			}
+			else if (lastOpponent != null
+				&& lastOpponent.getInteracting() != player)
+			{
+				lastTime = Instant.now();
+			}
 		}
+		else if (target == player
```

Comment:
Can the case of `target == player && opponent != null && opponent != lastOpponent` ever happen? You have 4 different branches here:

1) `target == player && opponent == null`
2) `target == player && opponent != null && opponent != lastOpponent` <-- does this happen?
3) `target != player && source == lastOpponent && target == player ...` obviously this cannot happen
4) `target != player && source == lastOpponent && opponent != lastOpponent`

I think this can't happen because it would require something to attack you while you were attacking something that also wasn't your opponent. And I don't think you can be attacking something which isn't your opponent.

If so, this can just be split I think into two separate ifs

```java
if (target == player && opponent == null) {
// Case of something attacking you when you aren't in combat
}
else if (source == lastOpponent && opponent != lastOpponent) {
// Case of something attacking you switching off while you are not attacking it
}
```